### DISCUSSION TITLE
Unable to add custom values to prefect.context

### DISCUSSION THREAD

**Discussion by sierrezinal** [PrefectHQ/prefect#3224] [Q&A] [answered]
```python
with Flow("myflow", schedule=schedule) as flow:
    jobid = Parameter("jobid")
    prefect.context.update({'jobid': jobid})
```

Later on in my task, reading from the context returns `None`.

**Comment by cicdw** [comment, marked as answer]
If you are using `flow.run`, the correct way to specify new context key / value pairs is via the `context` keyword argument:
```python
flow.run(context={'jobid': jobid})
```
If you are using Prefect Cloud or Server, the API allows you to provide new key / value pairs for each ad-hoc run.

Also, it appears that you are using context incorrectly.  You should never place tasks in context; if you are interested in passing data from one task to another, you should do so explicitly:
```python
with Flow("myflow", schedule=schedule) as flow:
    jobid = Parameter("jobid")
    another_task(jobid=jobid) # will pass the jobid value to the task
```

**Reply by sierrezinal** [reply]
```
flow.run(context={'jobid': jobid})
```

Aha. I need to RTFM more. Thanks.
